### PR TITLE
Add: Added hover animation desktop/mobile

### PR DESCRIPTION
**Pull request by HyunCafe**
I only added the code into the navigation.component.scss 
please review, its a long code because I had to make a separate desktop and mobile transitions for the transitions to lineup correctly with the text.

I think its simple enough but clean, and it matches the color scheme

### CODE REVIEW

**Comment by HyunCafe** [comment]
I also had to add some packages for angular source I believe, it might have added that to your files since I did it through the terminal (while installing typescript and angular, ran into alot of issues had to uninstall, reinstall, change paths, change myself to superuser, and add local packages but finally got it to work)

File: src/app/components/navigation/navigation.component.scss [hunk 1]
```diff
@@ -14,6 +14,77 @@
   font-weight: 500;
 }
 
+.navbar-desktop .navigation-section-link {
+  position: relative;
+
+  &:before {
```
Comment (ChelleAmon) [inline src/app/components/navigation/navigation.component.scss:20]: I do notice that there are some CSS properties that are repeated. We can use `@mixin`, which is a sass utility to define the styles that can be reused in the CSS file. Here is an example: 

```
@mixin exampleName($backgroundColor, $top){
    content: "";
    width: 0;
    height: 5px;
    background-color: $backgroundColor;
    position: absolute;
    top: $top;
    left: 0;
    transition: all 0.5s; 
}
```
and then use `@include` to call the @mixin function and pass the arguments

Here is the reference: https://sass-lang.com/documentation/at-rules/mixin

Also, I implemented @mixin  and @include in this file (check lines 10-15, and then lines 99 and 132).

**Review by ChelleAmon — COMMENTED** [review]
Great job and thank you once again for helping me out. I have a suggestion left. This is my first time merging a PR from a forked repo, so please give me some time to learn how to safely do it

File: src/app/components/navigation/navigation.component.scss (hunk 1 above)
Comment (HyunCafe) [reply src/app/components/navigation/navigation.component.scss:20]: Great, I will look into this and make the appropriate changes (have been meaning to learn sass next anyway)

and no problem, please take your time

**Comment by HyunCafe** [comment]
Alright, I made the required changes, I think it roughly saved about 20 lines of code, do I open up a new pull request? I am not sure how this works as well

File: src/app/components/navigation/navigation.component.scss (hunk 1 above)
Comment (HyunCafe) [reply src/app/components/navigation/navigation.component.scss:20]: Alright, I made the required changes, I think it roughly saved about 20 lines of code, do I open up a new pull request? I am not sure how this works as well

Comment (ChelleAmon) [reply src/app/components/navigation/navigation.component.scss:20]: No need to create a new PR. The changes you have made are already saved in this PR. This looks good to me!

**Review by ChelleAmon — DISMISSED** [review]

**Comment by HyunCafe** [comment]
perhaps I can just input the CSS changes directly through Github instead and bypass the changes I needed to do for the jsons for it to work on my local repo?

**Review by ChelleAmon — APPROVED** [review]

**Comment by ChelleAmon** [comment]
> perhaps I can just input the CSS changes directly through Github instead and bypass the changes I needed to do for the jsons for it to work on my local repo?

Hey! I am a bit confused about the question. Is the project no longer running on your local? 

I am going to merge the changes now. Thank you for your patience here!

**Pull request merged**